### PR TITLE
Added namespace support

### PR DESCRIPTION
**Pull request by fotomerchant**
This is an updated pull request with similar work to what was done in #20.

### CODE REVIEW

**Comment by f2prateek** [comment]
I'm still not sure why this change was made https://github.com/segmentio/analytics-php/pull/58#discussion_r81909151.

This is also a bit out of date now so I'm closing this for now.

**Comment by enkay** [comment]
Is this library maintained by the Segment team or by the community? It seems like every PR that tries to add namespacing gets closed without getting merged.

File: lib/Segment.php [hunk 1]
```diff
@@ -1,7 +1,9 @@
 <?php
 
+namespace Segment;
+
 if (!function_exists('json_encode')) {
-    throw new Exception('Segment needs the JSON PHP extension.');
+    throw new \Exception('Segment needs the JSON PHP extension.');
```
[the remark this answers is not in the document]
Comment (enkay) [reply lib/Segment.php]: When a namespace is added to the file, you need to reference the global namespace for the Exception. That's what the \ is for.

**Comment by f2prateek** [comment]
It's still maintained by segment, but do keep in mind that we're not day to day PHP developers so ultimately we rely heavily on community contributors. All previous attempts to address the namespace issue (including this one) have closed for specific reasons that we've explained as a comment.

**Pull request closed without merging**